go-controller/pkg/util/sriovnet_linux.go

```diff
@@ -118,3 +119,29 @@ func GetFunctionRepresentorName(deviceID string) (string, error) {
 	}
 	return rep, nil
 }
+
+// GetNetdevNameFromDeviceId returns the netdevice name from the passed device ID.
+func GetNetdevNameFromDeviceId(deviceId string) (string, error) {
+	var netdevices []string
+	var err error
+
+	if IsPCIDeviceName(deviceId) {
+		// If a vDPA device exists, it takes preference over the vendor device, steering-wize
+		var vdpaDevice kvdpa.VdpaDevice
+		vdpaDevice, err = GetVdpaOps().GetVdpaDeviceByPci(deviceId)
+		if err == nil && vdpaDevice.Driver() == kvdpa.VirtioVdpaDriver {
+			return vdpaDevice.VirtioNet().NetDev(), nil
+		}
+
+		netdevices, err = GetSriovnetOps().GetNetDevicesFromPci(deviceId)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	// Make sure we have 1 netdevice per pci address
+	if len(netdevices) != 1 {
+		return "", fmt.Errorf("failed to get one netdevice interface per Device ID %s", deviceId)
```

Comment:
add now many we have. It's easier to debug in the future if we get an error message that not only tells us that we don't have 1, but also how many we actually have.